index.js

```diff
@@ -55,8 +55,8 @@ class ServerlessCustomDomain {
 
     const createDomainName = this.getCertArn().then(data => this.createDomainName(data));
     return Promise.all([createDomainName])
-      .then(values => this.changeResourceRecordSet(values[0], 'CREATE'))
-      .then(() => (this.serverless.cli.log('Domain was created, may take up to 40 mins to be initialized.')))
+      .then(values => this.changeResourceRecordSet(values[0], 'UPSERT'))
+      .then(() => (this.serverless.cli.log(`${this.givenDomainName} was created/updated. New domains may take up to 40 minutes to be initialized.`)))
       .catch((err) => {
         throw new Error(`${err} ${this.givenDomainName} was not created.`);
```

Comment:
There is a possibility here that the domain will be created but you well get an error from this.changeResourceRecordSet function, so the domain is actually created but the error message will say it was not. 
I think this is the reason this issue was open: https://github.com/amplify-education/serverless-domain-manager/issues/44 

You can move this catch between createDomainName and changeResourceRecordSet or make each function throw specific error messages and decide by it what the catch at the end will send to the log.

btw, I suggest using async await, it's already supported in the LTS and make the code much cleaner

Reply:
I think you're right in that we can explain it better, but the error in #44 isn't _entirely_ wrong since it's created in API Gateway, but it doesn't get created as a resource record.

I'm also not opposed to using `async`/`await` in general but we've stuck to promises since it would require transpiling for versions of node as low as 4.0.